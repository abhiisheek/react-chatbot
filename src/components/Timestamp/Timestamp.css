@import '../_variables.css';
@import '../_colors.css';

.timestamp {
  font-size: var(--font-size-very-tiny);
  color: var(--timestamp-text-color);
  padding: 3px 0;
}