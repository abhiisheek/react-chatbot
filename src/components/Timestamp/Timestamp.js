import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import { getFormatedTime } from '../utils/DateTime';
import styles from './Timestamp.css';

/** Displays current time according to the given format & locale */
class Timestamp extends PureComponent {
  render() {
    const { locale, formatOptions } = this.props;
    return (
      <div className={styles.timestamp}>
        {getFormatedTime(new Date(), locale, formatOptions)}
      </div>
    );
  }
}

Timestamp.displayName = 'Timestamp';

Timestamp.propTypes = {
  /** Name of the locale */
  locale: PropTypes.string,
  /** Formatting options */
  formatOptions: PropTypes.object
};

Timestamp.defaultProps = {
  locale: 'en-US',
  formatOptions: {
    hour: '2-digit',
    minute: '2-digit',
    hour12: true
  }
};

export default Timestamp;
