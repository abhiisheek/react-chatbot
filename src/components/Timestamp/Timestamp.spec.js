import React from 'react';
import renderer from 'react-test-renderer';

import Timestamp from './Timestamp';

jest.mock('../utils/DateTime');

describe('Timestamp', () => {
  it('Should have displayName set on the component', () => {
    expect(Timestamp.displayName).toBeDefined();
  });

  it('Snapshot test', () => {
    const snapshot = renderer.create(<Timestamp />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
