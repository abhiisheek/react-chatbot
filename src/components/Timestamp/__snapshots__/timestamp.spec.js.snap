// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timestamp Snapshot test 1`] = `
<div
  className="timestamp"
>
  10:11 PM
</div>
`;
